cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -213,6 +213,43 @@ spec:
     k8s-app: kube-dns
 `
 
+	// CoreDNSService is the coredns Service manifest
+	CoreDNSService = `
```

Comment:
as far as i know, the reason for the re-use is because the kube-dns service is still used for CoreDNS.

Reply:
yes, I've kept the service name to be `kube-dns`, and only changed the annotation- this could be pretty silly.

Reply:
I'd prefer to slightly change current `KubeDNSService` template. And you can try to rename `KubeDNSService` to `DNSService` if you feel uncomfortable in creating coredns service.